### PR TITLE
Move CLI execution to runner

### LINKED ISSUE #1: Update registry-client to follow similar python CLI tool installations and executions
### 💡 Description

Right now, to run the software, you have to run:

```
./cli/makesignedrequest
```

We should make this pip installable so we 

### ⚔️ Parent Epic / Related Tickets

_No response_

### PR DESCRIPTION
**Pull request by jordanpadams**
* Add installation documentation to online docs
* Update setup.cfg with appropriate dependencies
* Upgrade sphinx
* Fixes per code style pre-commit issues

Resolves #1

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -23,60 +23,77 @@ Additional functionality may be built out in the future.
...
+    registry-client --help
     ```
+### Windows Installation
+For help installing Python, [see this documentation from PDS Deep Archive](https://nasa-pds.github.io/deep-archive/installation/index.html#installing-python-for-windows).
 
----
+To install the Registry Client software on Windows comprises the following steps:
 
-## BEGIN BOILERPLATE
+Installing Python 3.11 for Windows
```
Comment (nutjob4life) [inline README.md:51]: I think we need either bullets `-` or numbers `1.` for each line here; without such annotation, Markdown just runs all these together into a single paragraph in HTML:
Screenshot 2024-07-09 at 3 40 14 PM

File: README.md [hunk 2]
```diff
@@ -23,60 +23,77 @@ Additional functionality may be built out in the future.
...
+Creating a “virtual environment” to contain an isolated instance of Python 3.11
+Installing LXML 4.9.0 for Python 3.11 into the virtual environment
+Installing the PDS Deep Archive into the virtual environment
```
Comment (nutjob4life) [inline README.md:54]: Is the Deep Archive required in order to run the Registry Client?

File: README.md [hunk 3]
```diff
@@ -23,60 +23,77 @@ Additional functionality may be built out in the future.
...
+```
 
-    (put your run commands here)
+Or on Windows in PowerShell:
 
-[//]: # (TODO: POPULATE THESE TEAMS)
-## 🏃 Getting Started With This Template 
+```
+.\pds\Scripts\activate.ps1
+pip install --upgrade pds.deeparchive
```
Comment (nutjob4life) [inline README.md:68]: Should probably be: `pip install --upgrade pds.registry-client`

File: README.md [hunk 4]
```diff
@@ -23,60 +23,77 @@ Additional functionality may be built out in the future.
...
----
+2. Create a virtual environment
+    ```
+    python -m venv venv
+    source ./venv/bin/activate
+    ```
 
+3. Install the tool to the virtual environment
+    ```
+    pip install .
```
Comment (nutjob4life) [inline README.md:90]: Should probably be `pip install --editable .[dev]`

File: docs/source/installation/index.rst [hunk 5]
```diff
@@ -0,0 +1,235 @@
...
+      mkdir -p $HOME/.virtualenvs
+      python3 -m venv $HOME/.virtualenvs/pds-registry-client
+      source $HOME/.virtualenvs/pds-registry-client/bin/activate
+      pip3 install --index-url https://test.pypi.org/simple --extra-index-url https://pypi.org/simple`` pds.registry-client
+
+
+Windows Installation
+--------------------
+
+To install the Deep Archive software on Windows comprises the following steps:
```
Comment (nutjob4life) [inline docs/source/installation/index.rst:104]: "To install the Registry Client"

File: setup.cfg [hunk 6]
```diff
@@ -71,23 +72,23 @@ dev =
...
+    #pydocstyle~=6.3.0
     coverage~=7.3.0
     pytest~=7.4.0
     pytest-cov~=4.1.0
     pytest-watch~=4.2.0
     pytest-xdist~=3.3.1
     pre-commit~=3.3.3
-    sphinx~=3.2.1
-    sphinx-rtd-theme~=0.5.0
+    sphinx~=7.3.7
```
Comment (nutjob4life) [inline setup.cfg:82]: Might double-check this; the `github-actions-base` may force an older version of Sphinx to be used with the Roundup Action

**Review by nutjob4life — DISMISSED** [review]
Code looks good, just need a few doc updates I think

File: README.md (hunk 2 above)
Comment (jordanpadams) [reply README.md]: --no-verify

File: README.md (hunk 4 above)
Comment (jordanpadams) [reply README.md]: Fixed

File: setup.cfg (hunk 6 above)
Comment (jordanpadams) [reply setup.cfg:82]: @nutjob4life hmmmm. we will see. it seems like there was some bug happening that was preventing me from building the docs locally. we will see if it works in production.

Comment (nutjob4life) [reply setup.cfg:82]: @jordanpadams sounds good, fingers crossed 🤞

File: docs/source/installation/index.rst (hunk 5 above)
Comment (jordanpadams) [reply docs/source/installation/index.rst]: fixed

File: README.md (hunk 3 above)
Comment (jordanpadams) [reply README.md]: fixed

**Comment by jordanpadams** [comment]
@nutjob4life any idea what is going on here? https://github.com/NASA-PDS/registry-client/actions/runs/9864614468/job/27239892352#step:5:482

**Comment by nutjob4life** [comment]
> @nutjob4life any idea what is going on here? https://github.com/NASA-PDS/registry-client/actions/runs/9864614468/job/27239892352#step:5:482

Fixed the `mypy` problems in [8aef616](https://github.com/NASA-PDS/registry-client/pull/2/commits/8aef61642fe758ead5d065b23f9ddd079d8ab4d8) … but now secrets detection is failing 🤷

**Comment by alexdunnjpl** [comment]
Sorry gents - I hadn't gotten this to the "proper minimal project" stage yet

**Comment by nutjob4life** [comment]
@jordanpadams okay, I got (yet another) meeting before the day is over but I think the remaining issue is this: `tox.ini` wants to generate documentation by running `python setup.py build_sphinx` but that's for older Sphinxes.

I think if you change that to `sphinx-build -a -b html docs/source docs/build` you should be good to go

**Review by tloubrieu-jpl — APPROVED** [review]
Thanks @jordanpadams, you made the last step to make it from working tool to user ready tool.

**Comment by jordanpadams** [comment]
@nutjob4life it worked! 🎉

**Pull request merged**